injector/src/test/java/edu/ucr/cs/riple/injector/AnnotationWithArgumentTest.java

```diff
@@ -177,10 +181,204 @@ public void onParameterCustomAnnotationTest() {
+        .start();
+  }
+
+  @Test
+  public void onMethodCollapseOffTest() {
+    injectorTestHelper
+        .addInput(
+            "Super.java",
+            "package com.edu;",
+            "public class Super {",
+            "   Object h = new Object();",
+            "   @CustomNull({\"arg1\", \"arg2\"})",
+            "   public void test(Object f) {",
+            "      h = f;",
+            "   }",
+            "}")
+        .expectOutput(
+            "package com.edu;",
+            "import edu.ucr.CustomNull;",
+            "public class Super {",
+            "   Object h = new Object();",
+            "   @CustomNull({\"arg1\", \"arg2\"})",
+            "   @CustomNull(\"arg3\")",
+            "   public void test(Object f) {",
+            "      h = f;",
+            "   }",
+            "}")
+        .addChanges(
+            new AddSingleElementAnnotation(
+                new OnMethod("Super.java", "com.edu.Super", "test(java.lang.Object)"),
+                "edu.ucr.CustomNull",
+                "arg3",
+                false))
+        .start();
+  }
+
+  @Test
+  public void testImportNotExistsCollapseOff() {
+    injectorTestHelper
+        .addInput(
+            "Super.java",
+            "package com.edu;",
+            "public class Super {",
+            "   Object h = new Object();",
+            "   public void test(Object f) {",
+            "      h = f;",
+            "   }",
+            "}")
+        .expectOutput(
+            "package com.edu;",
+            "import edu.ucr.CustomNull;",
+            "public class Super {",
+            "   Object h = new Object();",
+            "   @CustomNull(\"arg3\")",
+            "   public void test(Object f) {",
+            "      h = f;",
+            "   }",
+            "}")
+        .addChanges(
+            new AddSingleElementAnnotation(
+                new OnMethod("Super.java", "com.edu.Super", "test(java.lang.Object)"),
+                "edu.ucr.CustomNull",
+                "arg3",
+                false))
+        .start();
+  }
+
+  @Test
+  public void testImportNotExistsCollapseOn() {
+    injectorTestHelper
+        .addInput(
+            "Super.java",
+            "package com.edu;",
+            "public class Super {",
+            "   Object h = new Object();",
+            "   public void test(Object f) {",
+            "      h = f;",
+            "   }",
+            "}")
+        .expectOutput(
+            "package com.edu;",
+            "import edu.ucr.CustomNull;",
+            "public class Super {",
+            "   Object h = new Object();",
+            "   @CustomNull(\"arg3\")",
+            "   public void test(Object f) {",
+            "      h = f;",
+            "   }",
+            "}")
+        .addChanges(
+            new AddSingleElementAnnotation(
+                new OnMethod("Super.java", "com.edu.Super", "test(java.lang.Object)"),
+                "edu.ucr.CustomNull",
+                "arg3",
+                true))
```

Comment:
Appreciate the comprehensive test suite here, btw!

Reply:
Sure 🙂